vars/toolValidate.groovy

```diff
@@ -1,31 +1,45 @@
 import com.sap.piper.FileUtils
 import com.sap.piper.Version
+import com.sap.piper.Utils
+import com.sap.piper.ConfigurationLoader
+
 import hudson.AbortException
 
 
 def call(Map parameters = [:]) {
 
-    handlePipelineStepErrors (stepName: 'toolValidate', stepParameters: parameters) {
+    def stepName = 'toolValidate'
 
-        def tool = parameters.tool
-        def home = parameters.home
+    handlePipelineStepErrors (stepName: stepName, stepParameters: parameters) {
+
+        final script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
 
-        if (!tool) throw new IllegalArgumentException("The parameter 'tool' can not be null or empty.")
-        if (!home) throw new IllegalArgumentException("The parameter 'home' can not be null or empty.")
+        final Map configuration = ConfigurationLoader.generalConfiguration(script)
+
+        def utils = new Utils()
+        def tool = utils.getMandatoryParameter(parameters, 'tool')
+        def home = parameters.home
```

Comment:
The pattern is that the parameters are merged with the other configuration layers. Afterwards there is one configuration map. After the merge the parameters should not be used. See ```vars/mavenExecute```for reference.